core/src/config.rs

```diff
@@ -1,24 +1,82 @@
 use arrayref::array_ref;
 use ed25519_dalek::*;
 use failure::Error;
+use hcid::HcidEncoding;
+use lazy_static::lazy_static;
 use rand::{rngs::OsRng, Rng};
 use serde::*;
+use std::fmt;
+use std::str::FromStr;
+
 
 const SEED_SIZE: usize = 32;
 
-fn public_key_from_base64<'de, D>(deserializer: D) -> Result<PublicKey, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    String::deserialize(deserializer)
-        .and_then(|s| {
-            base64::decode_config(&s, base64::STANDARD_NO_PAD)
-                .map_err(|err| de::Error::custom(err.to_string()))
-        })
-        .map(|bytes| PublicKey::from_bytes(&bytes))
-        .and_then(|maybe_key| maybe_key.map_err(|err| de::Error::custom(err.to_string())))
+
+/// Admin... keys are just ed25591 signing keys w/ some different serialization; HcAc...
+#[derive(Debug, Clone)]
+pub struct AdminPublicKey(pub PublicKey);
+
+lazy_static! {
+    pub static ref HCAPK_CODEC: hcid::HcidEncoding =
+        HcidEncoding::with_kind("hca0").expect("Couldn't init hca0 hcid codec.");
```

Comment:
You keep bringing this change back in (https://github.com/Holo-Host/holo-config/pull/2#discussion_r329140092, https://github.com/Holo-Host/holo-config/pull/4#discussion_r332105570). Could you send it as a separate PR?

Reply:
Yes.  Philosophically, when we scatter 256-bit hashes throughout our code and output files, each with different semantic "meanings", but with identical encoding (eg. base-64) -- everything works great, as long as we remember "Hmm..  This particular hash was .. right, it in this file, with this particular property name, and if I recall that was an Admin public signing ed25519 key!  OK, let me decode it and recover one of those from it".

Or, we can glance at the encoded form, and say "HcA.. Right. An admin ed25591 signing key".  Same for Seed (BIP39), Agent ID pubkey (HcS..), Admin pubkey (HcA..), DHT commit hash (Qm.. Sha-256), ...

We have about 20 such semantic encodings prefixes left; unless we're in a length-constrained situation (eg. the SSL key URL length limit), why not make these things clear?

Even for our SSL certificate base-36 encoding, where log(2^256) / log(36) =~= 50 symbols; this leaves only 13 symbols for the rest of the URL; `.holohost.net` uses this up completely; I think using a shorter suffix, and establishing a multihash prefix such `Hh` or something, resulting in a <=63 symbol URL such as as `Hhabc123...xyz.holo.host` would at least give some semantic context for what this encoding means, while satisfying the SSL certificate requirement (and still retaining recoverability of the 256-bit value).

I'm willing to forego any of this discoverability for now, in the interests of time, but I think we need to revisit it before the end of Alpha.